### PR TITLE
[TIMOB-12192] Xcode 4.6 build error fixes and changes required for profiler

### PR DESCRIPTION
**Pull request by mstepanov**
[TIMOB-12192](https://jira.appcelerator.org/browse/TIMOB-12192)

### CODE REVIEW

File: TiCore/profiler/ProfileGenerator.cpp [hunk 1]
```diff
@@ -87,8 +87,8 @@ void ProfileGenerator::willExecute(TiExcState* callerCallFrame, const CallIdenti
         JAVASCRIPTCORE_PROFILE_WILL_EXECUTE(m_profileGroup, const_cast<char*>(name.data()), const_cast<char*>(url.data()), callIdentifier.m_lineNumber);
     }
 
-    if (!m_origin)
-        return;
+    //if (!m_origin)
+    //    return;
```
Comment (WhichKatieDid) [inline TiCore/profiler/ProfileGenerator.cpp:91]: Out of curiosity, why do we disable these checks? Are they no longer valid checks?

**Comment by WhichKatieDid** [comment]
Beyond that curiosity, CR looks good.

File: TiCore/profiler/ProfileGenerator.cpp (hunk 1 above)
Comment (mstepanov) [inline TiCore/profiler/ProfileGenerator.cpp:91]: ProfileGenerator is created with NULL as context, allowing to collect stats across all available contexts.
Have no idea why they did the NULL-check in the first place.

**Comment by mstepanov** [comment]
@BlainHamon replied.

**Comment by WhichKatieDid** [comment]
FR passed. Merging.

**Pull request merged**